xdg-ninja.sh

```diff
@@ -30,58 +28,115 @@ fi
 
-    \033[3m--help\033[0m              \033[1mThis help menu\033[0m
-    \033[3m-h\033[0m
+        ────────────────────────────────────
 
-    \033[3m--no-skip-ok\033[0m        \033[1mDisplay messages for all files checked (verbose)\033[0m
-    \033[3m-v\033[0m
+        ${ANSI_BEGIN}${ANSI_ITALLIC}${ANSI_END}--help${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}              ${ANSI_BEGIN}${ANSI_BOLD}${ANSI_END}This help menu${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}
+        ${ANSI_BEGIN}${ANSI_ITALLIC}${ANSI_END}-h${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}
 
-    \033[3m--skip-ok\033[0m           \033[1mDon't display anything for files that do not exist (default)\033[0m
+        ${ANSI_BEGIN}${ANSI_ITALLIC}${ANSI_END}--no-skip-ok${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}        ${ANSI_BEGIN}${ANSI_BOLD}${ANSI_END}Display messages for all files checked (verbose)${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}
+        ${ANSI_BEGIN}${ANSI_ITALLIC}${ANSI_END}-v${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}
+
+        ${ANSI_BEGIN}${ANSI_ITALLIC}${ANSI_END}--skip-ok${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}           ${ANSI_BEGIN}${ANSI_BOLD}${ANSI_END}Don't display anything for files that do not exist (default)${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}
+        ${ANSI_BEGIN}${ANSI_ITALLIC}${ANSI_END}--skip-warn${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}         ${ANSI_BEGIN}${ANSI_BOLD}${ANSI_END}Don't display anything for files that cannot be fixed.${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}
+        ${ANSI_BEGIN}${ANSI_ITALLIC}${ANSI_END}--color=WHEN${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}        ${ANSI_BEGIN}${ANSI_BOLD}${ANSI_END}Color the output always, never, or auto (default)${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}
+        ${ANSI_BEGIN}${ANSI_ITALLIC}${ANSI_END}--decoder=DECODER${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}   ${ANSI_BEGIN}${ANSI_BOLD}${ANSI_END}Manually set the decoder used for markdown.${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}
+        ${ANSI_BEGIN}${ANSI_ITALLIC}${ANSI_END}--json${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}              ${ANSI_BEGIN}${ANSI_BOLD}${ANSI_END}Output json${ANSI_BEGIN}${ANSI_CLEAR}${ANSI_END}
 
 """
+printf "%b" "$HELPSTRING"
+exit
+}
 
 SKIP_OK=true
+SKIP_WARN=false
+COLOR=auto
+JSON=false
 for i in "$@"; do
-    if [ "$i" = "--help" ] || [ "$i" = "-h" ]; then
-        printf "%b" "$HELPSTRING"
-        exit
-    elif [ "$i" = "--skip-ok" ]; then
-        SKIP_OK=true
-    elif [ "$i" = "--no-skip-ok" ]; then
-        SKIP_OK=false
-    elif [ "$i" = "-v" ]; then
-        SKIP_OK=false
-    fi
+    case $i in
+        --color=*)
+            COLOR="${i#*=}"
+            ;;
+        --help|-h)
+            help
+            ;;
+        --skip-ok)
+            SKIP_OK=true
+            ;;
+        --skip-warn)
+            SKIP_WARN=true
+            ;;
+        --no-skip-ok|-v)
+            SKIP_OK=false
+            ;;
+        --decoder=*)
+            DECODER="${i#*=}"
+            ;;
+        --json)
```

Comment:
And an option for disabling json output, maybe (`--render`, `--human-readable`?)

Reply:
`--text`?

Reply:
I'll just go with `--no-json` for simplicity sake

Reply:
> I'll just go with `--no-json` for simplicity sake

But if there will be other output modes?

Reply:
> But if there will be other output modes?

Those probably won't, but I nevertheless think `--text` fits better here